app/controllers/actions/IdentifierAction.scala

```diff
@@ -53,12 +53,20 @@ class AuthenticatedIdentifierAction @Inject()(override val authConnector: AuthCo
           block(AdministratorRequest(affGroup, internalId, psaId.value, UserType.PSA, request))
         case Some(internalId) ~ Some(affGroup) ~ IsPSP(pspId) if hasValidSession(hc) =>
           block(PractitionerRequest(affGroup, internalId, pspId.value, UserType.PSP, request))
-        case Some(_) ~ Some(_) ~ _ => Future.successful(Redirect(config.loginUrl))
-        case _ => Future.successful(Redirect(routes.UnauthorisedController.onPageLoad()))
+        case Some(_) ~ Some(_) ~ _ if !hasValidSession(hc) =>
```

Comment:
I think this should be a redirect to auth login rather than to the session timeout page, see - https://github.com/hmrc/income-tax-tailor-returns-frontend/blob/main/app/utils/SessionIdHelper.scala#L13 for example